components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -119,346 +120,159 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	// Update namespace
 	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: instance.Name,
+			Annotations: map[string]string{"owner": instance.Spec.Owner.Name},
+			Name:        instance.Name,
 		},
 	}
 	if err := controllerutil.SetControllerReference(instance, ns, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 	foundNs := &corev1.Namespace{}
 	err = r.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, foundNs)
-	if err != nil && errors.IsNotFound(err) {
-		log.Info("Creating Namespace: " + ns.Name)
-		err = r.Create(context.TODO(), ns)
-		if err != nil {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating Namespace: " + ns.Name)
+			err = r.Create(context.TODO(), ns)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		} else {
 			return reconcile.Result{}, err
 		}
-	} else if err != nil {
-		return reconcile.Result{}, err
+	} else {
+		// Check exising namespace ownership before move forward
+		val, ok := foundNs.Annotations["owner"]
+		if (!ok) || val != instance.Spec.Owner.Name {
+			log.Info(fmt.Sprintf("namespace already exist, but not owned by profile creator %v",
+				instance.Spec.Owner.Name))
+			instance.Status = kubeflowv1alpha1.ProfileStatus{
+				Status: kubeflowv1alpha1.ProfileFailed,
+				Message: fmt.Sprintf("namespace already exist, but not owned by profile creator %v",
+					instance.Spec.Owner.Name),
+			}
+			r.Update(context.TODO(), instance)
+			return reconcile.Result{}, nil
+		}
 	}
-	// No need to update namespace
 
-	role := generateRole(instance)
-	if err := controllerutil.SetControllerReference(instance, role, r.scheme); err != nil {
-		return reconcile.Result{}, err
+	// Update service accounts
+	// "default-editor" would be default "edit" permission for pods in user namespace
```

Comment:
Could you clarify the comment, what's default-editor, what's edit, what does this function do